apps/shop/views.py

```diff
@@ -125,3 +114,37 @@ def list(self, request):
             raise APIException
         serializer = ShopContactsSerializer(queryset)
         return Response(serializer.data)
+
+
+class ReviewViewSet(viewsets.ViewSet):
+
+    def reviews_by_item(self, request, item_id=None):
+        """Return all comments belonging to item with given ID."""
+        try:
+            reviews = ItemModel.objects.get(id=item_id).get_reviews()
+        except (ObjectDoesNotExist, ValueError):
+            raise NotFound(detail="item not found")
+        serializer = ReviewSerializer(reviews, many=True)
+        return Response(serializer.data)
+
+    def add_review_by_item(self, request, item_id=None):
+        """Add a comment for item with given ID.
```

Comment:
Isn't it better to make default create method instead of add_review_by_item?
And what by_item means? We just add a review, and it is always correlated to item

Reply:
{'get': 'reviews_by_item', 'post': 'add_review_by_item'} 
the first method returns reviews, the second method creates review for unauthorized users.
but it can be done differently

Reply:
Тоді другий метод і назвати краще по типу "create_no_auth" або щось схоже